libs/common/src/state-migrations/migrations/3-fix-premium.ts

```diff
@@ -0,0 +1,48 @@
+// eslint-disable-next-line import/no-restricted-paths -- Used for token decoding, which are valid for days. We want the latest
+import { TokenService } from "../../auth/services/token.service";
+import { MigrationHelper } from "../migration-helper";
+import { Migrator, IRREVERSIBLE, Direction } from "../migrator";
+
+type ExpectedAccountType = {
+  profile?: { hasPremiumPersonally?: boolean };
+  tokens?: { accessToken?: string };
+};
+
+export class FixPremiumMigrator extends Migrator<2, 3> {
+  async migrate(helper: MigrationHelper): Promise<void> {
+    const accounts = await helper.getAccounts<ExpectedAccountType>();
+
+    async function fixPremium(userId: string, account: ExpectedAccountType) {
+      if (account?.profile?.hasPremiumPersonally === null && account.tokens?.accessToken != null) {
+        let decodedToken: { premium: boolean };
+        try {
+          decodedToken = await TokenService.decodeToken(account.tokens.accessToken);
+        } catch {
+          return;
+        }
+
+        if (decodedToken?.premium == null) {
+          return;
+        }
+
+        account.profile.hasPremiumPersonally = decodedToken?.premium;
+        return helper.set(userId, account);
+      }
+    }
+
+    await Promise.all(accounts.map(async ({ userId, account }) => fixPremium(userId, account)));
```

Comment:
nit: Unnecessary async (and in subsequent migrations)

```suggestion
    await Promise.all(accounts.map(({ userId, account }) => fixPremium(userId, account)));
```